src/core/options-manager.ts

```diff
@@ -113,13 +113,22 @@ class OptionsManager {
     }
 
     const { value, type } = valueDescriptor;
-    if (type === ValueType.Complex && value instanceof Object) {
+    if (type === ValueType.Complex && value instanceof Object
+        && !(value instanceof Array)
+    ) {
       Object.keys(value).forEach((key) => {
-        if (value[key] === (e.value as Record<string, unknown>)[key]) {
+        if (value[key] === (e.value as Record<string, unknown>)?.[key]) {
           return;
         }
         this.setGuard(mergeNameParts(e.fullName, key), value[key]);
       });
+    } else if (value instanceof Array && type === ValueType.Array) {
+      for (let i = 0; i < value.length; i += 1) {
+        if (value[i] !== (e.value as Array<unknown>)?.[i]) {
+          this.setGuard(e.fullName, [...value]);
```

Comment:
Do we need spread here? We pass this props to DevExtreme option manager, that update arrays non-reactive way.